armi/operators/tests/test_operators.py

```diff
@@ -76,5 +78,79 @@ def test_addInterfaceSubclassCollision(self):
         self.assertEqual(o.getInterface("Third"), interfaceC)
 
 
+class CyclesSettingsTests(unittest.TestCase):
+    """
+    Check that we can correctly access the various cycle settings from the operator.
+    """
+
+    detailedCyclesSettings = """
+metadata:
+  version: uncontrolled
+settings:
+  power: 1000000000.0
+  nCycles: 3
+  cycles:
+    - name: dog
+      cumulative days: [1, 2, 3]
+      power fractions: [0.1, 0.2, 0.3]
+      availability factor: 0.1
+    - cycle length: 10
+      burn steps: 5
+      power fractions: [0.2, 0.2, 0.2, 0.2, 0]
+      availability factor: 0.5
+    - name: ferret
+      step days: [3, R4]
+      power fractions: [0.3, R4]
+  runType: Standard
+"""
+
+    powerFractionsSolution = [
+        [0.1, 0.2, 0.3],
+        [0.2, 0.2, 0.2, 0.2, 0],
+        [0.3, 0.3, 0.3, 0.3, 0.3],
+    ]
+    cycleNamesSolution = ["dog", None, "ferret"]
+    availabilityFactorsSolution = [0.1, 0.5, 1]
+    stepLengthsSolution = [
+        [1, 1, 1],
+        [10 / 5 * 0.5, 10 / 5 * 0.5, 10 / 5 * 0.5, 10 / 5 * 0.5, 10 / 5 * 0.5],
+        [3, 3, 3, 3, 3],
+    ]
+    cycleLengthsSolution = [30, 10, 15]
+    burnStepsSolution = [3, 5, 5]
+    maxBurnStepsSolution = 5
+
+    def setUp(self):
+        self.standaloneDetailedCS = Settings()
+        self.standaloneDetailedCS.loadFromString(self.detailedCyclesSettings)
+        self.detailedOperator = Operator(self.standaloneDetailedCS)
+
+    def test_getPowerFractions(self):
+        self.assertEqual(
+            self.detailedOperator.powerFractions, self.powerFractionsSolution
+        )
+
+    def test_getCycleNames(self):
+        self.assertEqual(self.detailedOperator.cycleNames, self.cycleNamesSolution)
+
+    def test_getAvailabilityFactors(self):
+        self.assertEqual(
+            self.detailedOperator.availabilityFactors,
+            self.availabilityFactorsSolution,
+        )
+
+    def test_getStepLengths(self):
```

Comment:
I love these tests!

I was wondering if we could add a line to all of these "getter" tests.  For instance, here, if you set:

```python
self.detailedOperator.stepLengths = None
```

And then you ran the same `self.assertEqual()` line again, you would expect to get the same result, right?

But, if you add these two extra lines we make sure to cover more of the code under the "getter" method you wrote by unit tests.

That's all, I'm just looking at the code coverage of all these getters you wrote, and since you start with a fully complete operator, your tests don't flex the `if not self._stepLengths:` parts of these methods.

Reply:
Done.